Update PPP for mjgui (#211)
AY2223S1-CS2103T-W13-4/tp: docs/team/mjgui.md

---

> layout: page title: Ming Jiang's Project Portfolio Page

Reviewer: I'm guessing this is unintended

Given below are my contributions to the project.

> * **New Feature**: `assign task` - Added command to assign tasks randomly.

Reviewer: Isn't this `assign random`?

    * **Highlights**: The `java.util.Random` class was used to generate a random user index to be assigned. Care was
      taken to first filter users that have not been assigned that task (so as not to assign the same user twice).

* **New Feature**: `summary` - Added command to see a quick summary of the tasks each member is working on.
    * **What it does**: Displays a breakdown of the number of tasks assigned to each person.
    * **Justification**: In a large project, it's useful to have an overview on the number of tasks each person is
      working on to ensure a reasonably fair distribution of work. However, manually counting the number of tasks done
      by each user is very time-consuming.
    * **Highlights**: For the implementation of this feature, a `HashMap` was used to keep track of the counts of tasks
      for each team member. As the assignees of each task are stored within each team member, a linear pass through the
      list of tasks and each of its members was done instead of a linear pass of tasks for each member. This
      implementation reduces the runtime of the feature to linear (i.e. `O(n)`) in terms of task-assignee pairs.

> * **New Feature**: Added completion percentage of tasks on the main page.

Reviewer: ```suggestion
* **New Feature**: Added completion percentage of tasks to the UI.
```

* **Documentation**:
    * User Guide:
        * Add details for `tasksof`, `summary` and `assign random` commands
        * Update inaccurate command syntax in User Guide highlighted during PE Dry Run
        * Reorder commands into sections
    * Developer Guide:
        * Updated Logic component, Model component and Storage component diagrams and descriptions
        * Added description of Assign Task Randomly feature
        * Reviewed and edited manual testing
        * Improved formatting (changed ordering of contents, fixed hierarchy of elements in table of contents)

* **Community**:
    * PRs reviewed (with non-trivial review comments): [\#75](https://github.com/AY2223S1-CS2103T-W13-4/tp/pull/75)
      , [\#88](https://github.com/AY2223S1-CS2103T-W13-4/tp/pull/88)

>       , [\#95](https://github.com/AY2223S1-CS2103T-W13-4/tp/pull/88)

Reviewer: ```suggestion
      , [\#95](https://github.com/AY2223S1-CS2103T-W13-4/tp/pull/95)
```
